Source/WebCore/css/CSSProperties.json

```diff
@@ -1509,7 +1509,12 @@
                     "border-bottom-width",
                     "border-left-color",
                     "border-left-style",
-                    "border-left-width"
+                    "border-left-width",
+                    "border-image-source",
+                    "border-image-slice",
+                    "border-image-width",
+                    "border-image-outset",
+                    "border-image-repeat"
```

Comment:
This is good, but in `StyleProperties::getPropertyValue` you should check that these new longhands are present and set to their initial values before serializing `CSSPropertyBorder`. Otherwise it would be bad for `asTextInternal()`

Reply:
Good point. I should probably add a test too.

Reply:
I decided to put this into `commonShorthandChecks`.

Reply:
I am really close to removing all the shorthand-specific versions of this since it’s now checked in `commonShorthandChecks` for almost all cases. And remove all the exceptions in `commonShorthandChecks`. I’ll do those in a follow-up because this change is getting big.

Reply:
Now `commonShorthandChecks` checks that these new longhands are present, but `borderPropertyValue` or `getPropertyValue` should check that they are set to the initial value

Reply:
Yes, that jogs my memory. I was adding support for all the border-image longhands to isInitialValueForLonghand and it was a tedious project because of the way they are always parsed into quads even when it’s a single value. I decided to not include that work in this pull request because it was already kind of large.

But I suppose if I don’t do all that work then this will create a regression, where `border` will serialize and the `border-image` value will be lost.

Reply:
Maybe just check that `getPropertyValue(CSSPropertyBorderImage)` is `"none"` or something like that?
Or don't add these longhands for now.

Reply:
I fixed it the "hard" way.